test/e2e/serial/workload_placement_test.go

```diff
@@ -507,6 +518,135 @@ var _ = Describe("[serial][disruptive][scheduler] workload placement", func() {
 			Expect(schedOK).To(BeTrue(), "pod %s/%s not scheduled with expected scheduler %s", updatedPod.Namespace, updatedPod.Name, schedulerName)
 		})
 	})
+
+	When("cluster has two feasible nodes with taint but only one has the requested resources on a single NUMA zone", func() {
+		timeout := 5 * time.Minute
+		BeforeEach(func() {
+			numOfNodeToBePadded := len(nrts) - 1
+
+			rl := corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("3"),
+				corev1.ResourceMemory: resource.MustParse("8G"),
+			}
+			By("padding the nodes before test start")
+			err := padder.Nodes(numOfNodeToBePadded).UntilAvailableIsResourceList(rl).Pad(timeout)
+			Expect(err).ToNot(HaveOccurred())
+
+			nodes, err := e2enodes.GetWorkerNodes(fxt.Client)
+			Expect(err).ToNot(HaveOccurred())
+
+			t, _, err := taints.ParseTaints([]string{testTaint()})
+			Expect(err).ToNot(HaveOccurred())
+
+			for i := range nodes {
+				node := &nodes[i]
+				updatedNode, updated, err := taints.AddOrUpdateTaint(node, &t[0])
+				Expect(err).ToNot(HaveOccurred())
+				if updated {
+					node = updatedNode
+					klog.Infof("adding taint: %q to node: %q", t[0].String(), node.Name)
+					err = fxt.Client.Update(context.TODO(), node)
```

Comment:
is not clear why if `AddOrUpdateTaint` succeeded... we need to update the node again?

Reply:
```AddOrUpdateTaint``` update the given object in memory. It doesn't perform a call for the API server.

Reply:
ok, makes sense